cypress/support/commands.ts

```diff
@@ -582,23 +582,31 @@ export const setupUser = (): Cypress.Chainable<Cypress.Response<any>> => {
       url: `/debug/provision${userParam}`,
     })
     .then(response => {
-      expect(response.status).to.eq(200)
-      Cypress.env('defaultUser', response.body.user.name)
-      return response
+      if (response.status === 200) {
+        Cypress.env('defaultUser', response.body.user.name)
+        return response
+      } else {
+        // if for some reason the user wasn't flushed, do it now
+        return cy.flush().then(_ => cy.setupUser().then(res => res))
```

Comment:
is this how recursion works in cypress? the "its always a promise except when..." logic gets me turned around sometimes

Reply:
I think as long as we're returning a `Cypress.Chainable` object we should be ok